app/helpers/hosts_and_hostgroups_helper.rb

```diff
@@ -75,7 +75,7 @@ def puppet_master(f, can_override, override)
     select_f f, :puppet_proxy_id, proxies, :id, :name,
              { :include_blank => blank_or_inherit_f(f, :puppet_proxy),
                :disable_button => can_override ? _(INHERIT_TEXT) : nil,
-               :disable_button_enabled => override && !explicit_value?(:puppet_proxy_id),
+               :disable_button_enabled => params["inherit_all"] || (override && !explicit_value?(:puppet_proxy_id)),
```

Comment:
I wonder, if all the methods `puppet_master_fields`, `puppet_master`, `puppet_ca` and `realm_field` all need to have the `override` attribute, and if we should not just use the `inherited_by_default` method here instead. @tbrisker do you happen to know the reasons?